assets/js/modules/analytics-4/components/dashboard/ActivationBanner/SetupBanner.js

```diff
@@ -19,25 +19,98 @@
 /**
  * WordPress dependencies
  */
-import { __ } from '@wordpress/i18n';
+import { __, sprintf } from '@wordpress/i18n';
 
 /**
  * Internal dependencies
  */
+import Data from 'googlesitekit-data';
 import BannerNotification from '../../../../../components/notifications/BannerNotification';
+import { Grid, Row, Cell } from '../../../../../material-components';
+import { MODULES_ANALYTICS_4 } from '../../../datastore/constants';
+const { useSelect } = Data;
 
 export default function SetupBanner( { onCTAClick } ) {
+	const ga4MeasurementID = useSelect( ( select ) =>
+		select( MODULES_ANALYTICS_4 ).getMeasurementID()
+	);
+	const existingTag = useSelect( ( select ) =>
+		select( MODULES_ANALYTICS_4 ).getExistingTag()
+	);
+
+	let title;
+	let ctaLabel;
+	let footer;
+
+	switch ( true ) {
+		case ! ga4MeasurementID && ! existingTag:
```

Comment:
Please, use `if ( ... ) { ... } else if ( ... ) { ... } else if ( ... ) { ... } ...` approach instead of `switch( true )` magic.